From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tu Dinh <ngoc-tu.dinh@vates.tech>
Date: Mon, 18 May 2026 14:49:55 +0200
Subject: [PATCH] remove-vlan-tag-from-a-packet


diff --git a/NetworkPkg/MnpDxe/MnpIo.c b/NetworkPkg/MnpDxe/MnpIo.c
index 087c879c46a3..9ff8f3220eab 100644
--- a/NetworkPkg/MnpDxe/MnpIo.c
+++ b/NetworkPkg/MnpDxe/MnpIo.c
@@ -913,14 +913,7 @@ MnpReceivePacket (
   }
 
   VlanId = 0;
-  if (MnpDeviceData->NumberOfVlan != 0) {
-    //
-    // VLAN is configured, remove the VLAN tag if any
-    //
-    IsVlanPacket = MnpRemoveVlanTag (MnpDeviceData, Nbuf, &VlanId);
-  } else {
-    IsVlanPacket = FALSE;
-  }
+  IsVlanPacket = MnpRemoveVlanTag (MnpDeviceData, Nbuf, &VlanId);
 
   MnpServiceData = MnpFindServiceData (MnpDeviceData, VlanId);
   if (MnpServiceData == NULL) {
