diff --git a/OvmfPkg/OvmfPkgX64.dsc b/OvmfPkg/OvmfPkgX64.dsc
index 67f7e155ee..ca4379331c 100644
--- a/OvmfPkg/OvmfPkgX64.dsc
+++ b/OvmfPkg/OvmfPkgX64.dsc
@@ -40,6 +40,7 @@
   DEFINE SMM_REQUIRE             = FALSE
   DEFINE TLS_ENABLE              = FALSE
   DEFINE TPM2_ENABLE             = FALSE
+  DEFINE IPXE_ENABLE             = FALSE
 
   #
   # Flash size selection. Setting FD_SIZE_IN_KB on the command line directly to
diff --git a/OvmfPkg/OvmfPkgX64.fdf b/OvmfPkg/OvmfPkgX64.fdf
index 2e2a1749b5..f2e1fa8506 100644
--- a/OvmfPkg/OvmfPkgX64.fdf
+++ b/OvmfPkg/OvmfPkgX64.fdf
@@ -305,6 +305,14 @@ INF MdeModulePkg/Logo/LogoDxe.inf
   FILE DRIVER = 5D695E11-9B3F-4b83-B25F-4A8D5D69BE07 {
     SECTION PE32 = Intel3.5/EFIX64/E3522X2.EFI
   }
+!endif
+!if $(IPXE_ENABLE) == TRUE
+  FILE DRIVER = 9C1011A2-FC75-4C61-BFF1-AFAACA86DCD0 {
+    SECTION PE32 = 10ec8139.efi
+  }
+  FILE DRIVER = C4F14A61-D11E-499E-AE53-3CC64428E55A {
+    SECTION PE32 = 8086100e.efi
+  }
 !endif
   INF  MdeModulePkg/Universal/Network/SnpDxe/SnpDxe.inf
   INF  MdeModulePkg/Universal/Network/DpcDxe/DpcDxe.inf
