diff --git a/OvmfPkg/OvmfXen.dsc b/OvmfPkg/OvmfXen.dsc
index e565d528217a..f97c58d3448e 100644
--- a/OvmfPkg/OvmfXen.dsc
+++ b/OvmfPkg/OvmfXen.dsc
@@ -32,6 +32,7 @@
   #
   DEFINE SOURCE_DEBUG_ENABLE     = FALSE
   DEFINE BGRT_ENABLE             = TRUE
+  DEFINE IPXE_ENABLE             = FALSE
 
 !include OvmfPkg/OvmfTpmDefines.dsc.inc
 
diff --git a/OvmfPkg/OvmfXen.fdf b/OvmfPkg/OvmfXen.fdf
index 1ede502bac61..8c064582f4cb 100644
--- a/OvmfPkg/OvmfXen.fdf
+++ b/OvmfPkg/OvmfXen.fdf
@@ -327,6 +327,14 @@ INF MdeModulePkg/Logo/LogoDxe.inf
     SECTION PE32 = Intel3.5/EFIX64/E3522X2.EFI
   }
 !endif
+!if $(IPXE_ENABLE) == TRUE
+  FILE DRIVER = 9C1011A2-FC75-4C61-BFF1-AFAACA86DCD0 {
+    SECTION PE32 = 10ec8139.efi
+  }
+  FILE DRIVER = C4F14A61-D11E-499E-AE53-3CC64428E55A {
+    SECTION PE32 = 8086100e.efi
+  }
+!endif
 !include NetworkPkg/Network.fdf.inc
 
 #
