diff --git a/OvmfPkg/Include/OvmfPlatforms.h b/OvmfPkg/Include/OvmfPlatforms.h
index f613dd7e2d..382db4be17 100644
--- a/OvmfPkg/Include/OvmfPlatforms.h
+++ b/OvmfPkg/Include/OvmfPlatforms.h
@@ -24,6 +24,15 @@
 #define OVMF_HOSTBRIDGE_DID \
   PCI_LIB_ADDRESS (0, 0, 0, PCI_DEVICE_ID_OFFSET)
 
+//
+// Intel GVT-g changes Device ID of the Host Bridge for its drivers.
+// Fake the original Device ID for OVMF.
+//
+#define PciRead16(Address) \
+ ((Address == OVMF_HOSTBRIDGE_DID && \
+   PciRead16(Address) != INTEL_Q35_MCH_DEVICE_ID) ? \
+       INTEL_82441_DEVICE_ID : PciRead16(Address))
+
 //
 // Values we program into the PM base address registers
 //
diff --git a/OvmfPkg/QemuVideoDxe/Driver.c b/OvmfPkg/QemuVideoDxe/Driver.c
index e4df1ee9f5..b5cbeb7218 100644
--- a/OvmfPkg/QemuVideoDxe/Driver.c
+++ b/OvmfPkg/QemuVideoDxe/Driver.c
@@ -75,6 +75,12 @@ QEMU_VIDEO_CARD  gQemuVideoCardList[] = {
     0,
     QEMU_VIDEO_BOCHS_NVIDIA,
     L"NVIDIA vGPU"
+  },{
+    PCI_CLASS_DISPLAY_VGA,
+    0x8086,
+    0x0,
+    QEMU_VIDEO_BOCHS_INTEL,
+    L"Intel GVT-g"
   },{
     0     /* end of list */
   }
@@ -93,7 +99,8 @@ QemuVideoDetect (
     if ((gQemuVideoCardList[Index].SubClass == SubClass) &&
         (gQemuVideoCardList[Index].VendorId == VendorId) &&
         ((gQemuVideoCardList[Index].DeviceId == DeviceId) ||
-         (gQemuVideoCardList[Index].VendorId == 0x10de)))
+         (gQemuVideoCardList[Index].VendorId == 0x10de) ||
+         (gQemuVideoCardList[Index].VendorId == 0x8086)))
     {
       return gQemuVideoCardList + Index;
     }
@@ -428,6 +435,7 @@ QemuVideoControllerDriverStart (
       break;
     case QEMU_VIDEO_BOCHS_MMIO:
     case QEMU_VIDEO_BOCHS_NVIDIA:
+    case QEMU_VIDEO_BOCHS_INTEL:
     case QEMU_VIDEO_BOCHS:
       Status = QemuVideoBochsModeSetup (Private, IsQxl);
       break;
@@ -1027,7 +1035,8 @@ InitializeBochsGraphicsMode (
     );
 
   SetDefaultPalette (Private);
-  if (Private->Variant != QEMU_VIDEO_BOCHS_NVIDIA)
+  if (Private->Variant != QEMU_VIDEO_BOCHS_NVIDIA &&
+      Private->Variant != QEMU_VIDEO_BOCHS_INTEL)
     ClearScreen (Private);
 }
 
diff --git a/OvmfPkg/QemuVideoDxe/Gop.c b/OvmfPkg/QemuVideoDxe/Gop.c
index 3ff71e2317..fbb45b84c4 100644
--- a/OvmfPkg/QemuVideoDxe/Gop.c
+++ b/OvmfPkg/QemuVideoDxe/Gop.c
@@ -66,7 +66,8 @@ QemuVideoCompleteModeData (
                     (VOID **)&FrameBufDesc
                     );
 
-  if (Private->Variant == QEMU_VIDEO_BOCHS_NVIDIA)
+  if (Private->Variant == QEMU_VIDEO_BOCHS_NVIDIA ||
+      Private->Variant == QEMU_VIDEO_BOCHS_INTEL)
     Mode->FrameBufferBase = ((EFI_PHYSICAL_ADDRESS)BochsRead (Private,
                                  VBE_DISPI_INDEX_LFB_ADDRESS_H)) << 16 |
                              BochsRead (Private, VBE_DISPI_INDEX_LFB_ADDRESS_L);
@@ -189,6 +190,7 @@ Routine Description:
       break;
     case QEMU_VIDEO_BOCHS_MMIO:
     case QEMU_VIDEO_BOCHS_NVIDIA:
+    case QEMU_VIDEO_BOCHS_INTEL:
     case QEMU_VIDEO_BOCHS:
       InitializeBochsGraphicsMode (Private, ModeData);
       break;
diff --git a/OvmfPkg/QemuVideoDxe/Qemu.h b/OvmfPkg/QemuVideoDxe/Qemu.h
index 6dce753f3d..3b3bb03076 100644
--- a/OvmfPkg/QemuVideoDxe/Qemu.h
+++ b/OvmfPkg/QemuVideoDxe/Qemu.h
@@ -86,6 +86,7 @@ typedef enum {
   QEMU_VIDEO_BOCHS,
   QEMU_VIDEO_BOCHS_MMIO,
   QEMU_VIDEO_BOCHS_NVIDIA,
+  QEMU_VIDEO_BOCHS_INTEL,
   QEMU_VIDEO_VMWARE_SVGA,
 } QEMU_VIDEO_VARIANT;
 
